frontend-thea-aa/src/lib/components/Map.svelte

```diff
@@ -1,55 +1,76 @@
-			// }).addTo(map)
-
-			if (props.emissionEvent) {
-				props.emissionEvent.forEach((event) => {
-					const marker = L.marker([event.location.lat, event.location.long]).addTo(map)
-					marker.bindPopup(`Emission Event: ${event.site}`).openPopup()
-				})
-			}
-			if (props.contaminatedSite) {
-				props.contaminatedSite.forEach((event) => {
-					const marker = L.marker([event.location.lat, event.location.long]).addTo(map)
-					marker.bindPopup(`Contaminated site: ${event.name}`).openPopup()
-				})
-			}
-		}
-	})
-</script>
-
-<link
-	rel="stylesheet"
-	href="https://unpkg.com/leaflet@1.9.4/dist/leaflet.css"
-	integrity="sha256-p4NxAoJBhIIN+hmNHrzRCf9tD/miZyoHS5obTRR9BMY="
-	crossorigin=""
-/>
+          subdomains: 'abcd',
+          maxZoom: 19
+        }).addTo(map)
+      }
+    })
 
-<div bind:this={mapElement} class="h-96 w-96"></div>
+    export function updateViewCentering(lat: number, long: number, radius: number) {
+        if (!map) return;
+        map.setView([lat, long], 8);
+    }
+  
+    export function updateMarkers() {
+      if (!map || !L) return;
+      
+      // Clear existing markers
+      map.eachLayer((layer: any) => {
+        if (layer instanceof L.Marker) {
+          map.removeLayer(layer);
+        }
+      });
+  
+      // Add markers based on current view
+      if (props.currentView === 'superfund' && props.contaminatedSite) {
+        props.contaminatedSite.forEach((site) => {
+          const marker = L.marker([site.location.lat, site.location.long]).addTo(map);
+          marker.bindPopup(`Contaminated Site: ${site.name}`).openPopup();
+        });
+      } else if (props.currentView === 'emission' && props.emissionEvent) {
+        props.emissionEvent.forEach((event) => {
+          const marker = L.marker([event.location.lat, event.location.long]).addTo(map);
+          marker.bindPopup(`Emission Event: ${event.site}`).openPopup();
+        });
+      } else if (props.currentView === 'air quality' && props.airQualitySite) {
+        props.airQualitySite.forEach((site) => {
+          const marker = L.marker([site.location.lat, site.location.long]).addTo(map);
+          marker.bindPopup(`Air Quality Site: ${site.station}`).openPopup();
+        });
+      }
+    }
+  
+    // Watch for props changes
+    $effect(() => {
+      if (map && (props.currentView || props.contaminatedSite || props.emissionEvent || props.airQualitySite)) {
+        updateMarkers();
+      }
+    });
+  </script>
+  
+  <link
+    rel="stylesheet"
+    href="https://unpkg.com/leaflet@1.9.4/dist/leaflet.css"
+    integrity="sha256-p4NxAoJBhIIN+hmNHrzRCf9tD/miZyoHS5obTRR9BMY="
+    crossorigin=""
+  />
+  <div bind:this={mapElement} class="h-96 w-96" style="width: {typeof window !== 'undefined' ? 0.6 * window.innerWidth + 'px' : '100%'}; height: {typeof window !== 'undefined' ? 0.7 * window.innerHeight + 'px' : '100%'};"></div>
```

Comment:
does this work when you resize, I am unsure of whether svelte is reactive to window inner height changes without a binding, it could work but also could not and it worth checking